MPS writer refactoring : defining an mps writer interface and consequences

guilpier-code opened the pull request:
Each way of printing MPS files (full historic, or-tools way, split format or none) is now isolated inside a different class. Each of these ways can be removed or changed without impacting the client code. A factory function contains the logic that leads to the creation of a particular MPS writer.

guilpier-code committed:
mps writer refactoring : defining an mps writer interface and consequences

Each way of printing MPS files (full historic, or-tools way, split format or none) is now isolated
inside a different class. Each of these ways can be removed or changed without impacting the client code.
A factory function contains the logic that leads to the creation of a particular MPS writer.

guilpier-code committed:
mps writer refactoring : fix compilation issue on Ubuntu

guilpier-code committed:
[skip ci] mps writer refactoring : add a simple comment

guilpier-code committed:
mps writer refactoring : fix a crash on tests

guilpier-code committed:
mps writer refactoring : enhancing the MPS writer factory (from functions to a class)

guilpier-code committed:
[skip ci] mps writer refactoring : remove commented coded

flomnes commented on src/solver/utils/mps_utils.h:
```diff
@@ -8,7 +8,104 @@ extern "C"
+private:
+	// Member functions...
+	std::unique_ptr<I_MPS_writer> createFullmpsWriter();
+	bool doWeExportMPS();
```
```suggestion
	bool doWeExportMPS() const;
```

guilpier-code committed:
MPS files printing : restore print of 1st and 2nd optimization, whether split or not   (#872)

* mps dumping 1st and 2nd optim : just a save (does not compile)

* mps dumping 1st and 2nd optim : end of functionality development (see following comments)

- GUI : the MPS export status has now 4 items, the user can choose one of them (no mps, mps for optim 1 or optim 2, or both).
- consequences in the solver

* [skip ci] mps dumping 1st and 2nd optim : cleaning + renaming

* mps dumping 1st and 2nd optim : take mps files new names into account in infeasible problem tests

* Make absence of pytest or numpy fatal at configure time (#850)

* [CMake] Make absence of pytest or numpy fatal at configure time

* Remove useless CMake parameter

* Use pip instead of pip3

* Do not setup python 3.x

* Use py 3.10

* Disable registry search in Python3

See https://cmake.org/cmake/help/latest/module/FindPython3.html

* Revert "Disable registry search in Python3"

This reverts commit 7e8a67d711458b1c6893ec6cd6f9f17f6f67e389.

* Remove Python 3.10_x86

* Python3_FIND_REGISTRY "LAST"

* Keep Python 3.10_x86

* Cache pip deps

* Fix windows-vcpkg.yml

* [Win] Also run tests on releases

* Remove 'pip' caching

* Rename step

* mps dumping 1st and 2nd optim : correction due to review

* mps dumping 1st and 2nd optim : making compilation pass on Linux

* mps dumping 1st and 2nd optim : corrections due to sonarQ review

* [ski ci] mps dumping 1st and 2nd optim : documentation

* mps dumping 1st and 2nd optim : corrections due to review

Co-authored-by: Florian Omnès <florian.omnes@rte-france.com>

flomnes committed:
Merge remote-tracking branch 'github/develop' into feature/write-mps-refactor

flomnes committed:
Rewrite mpsWriterFactory::doWeExportMPS (#888)

flomnes reviewed (changes requested):
Some tests still fail
